### DISCUSSION TITLE
[Solved] Garmin GPS does not play with Debian 12?

### DISCUSSION THREAD

**Discussion by MartinHemma** [SpiralLinux/SpiralLinux-project#439] [Q&A] [answered]
During the sailing season I use to transfer files from my GPS (Garmin GPSmap 60C) to my computer for analysis and illustrations. I have done this for many, many years.

Now I find this does not work on my new computer running Debian 12 (Spiral Linux). For some reason the GPS finds this computer objectionable some seconds after I connect the GPS cable: It beeps and the ikon indicating connection disappears from its screen.

When I try with my side-kick laptop (Lenovo X230 so some ten years of age) I get the same behavior. This laptop also runs Debian 12 (Spiral Linux).

In both cases Linux still thinks the GPS is there (according to dmesg) but file transfer is not possible.

When I try with my old (13 years I think) computer running Debian 10 I have no problem connecting the GPS and transferring files.

I see no obvious difference between these machines when comparing the output from dmesg and lsusb after having connected the GPS .

Searching the net has so far not yielded anything useful.

Any ideas? It would be nice to not be dependent on that old computer.

TIA,

/Martin

**Comment by geckolinux** [comment]
Hi there, could you please try booting with the Debian 12 Stable kernel (version 6.1.0) that is also installed by default with SpiralLinux?

**Reply by MartinHemma** [reply]
Same behavior with 6.1.0 kernel :-(

/Martin

**Comment by AlphaElwedritsch** [comment]
Maybe missing MTP packages?

![image](https://github.com/SpiralLinux/SpiralLinux-project/assets/109347938/acb70958-96ec-4cca-899c-9f3197991cbb)

**Reply by MartinHemma** [reply]
Checked this and was missing jmtpfs and mtp-tools compared to your list. I installed them to no avail :-(

Following these instruction everything seems OK but my GPS does not agree. It prefers my old Debian 10 box.
https://wiki.openstreetmap.org/wiki/USB_Garmin_on_GNU/Linux

/Martin

**Comment by MartinHemma** [comment, marked as answer]
I have a solution now. Thanks to a Debian group at usenet news I learned about disabling automatic USB power save. Before I hook up my GPS to my new computer I use this command:

`sudo sh -c 'echo -1 > /sys/module/usbcore/parameters/autosuspend'`

/Martin

**Comment by Makalipay** [comment]
Permanent fix:

sed -i 's/GRUB_CMDLINE_LINUX_DEFAULT="/&usbcore.autosuspend=-1 /' /etc/default/grub

update-grub

systemctl reboot

**Reply by geckolinux** [reply]
You can also look into the TLP settings in `/etc/tlp.conf` which sets a lot of those autosuspend powersaving parameters.

**Reply by MartinHemma** [reply]
AmotanLtd's permanent fix works like a charm and makes sense to use as this is a stationary computer.

I have peeked into `/etc/tlp.conf` but I am not sure what to make of it since every line is commented out.

/Martin

**Reply by geckolinux** [reply]
Hi there. Whether it's a desktop or a laptop computer SpiralLinux includes TLP out-of-the-box, and that is what enables the powersaving parameters by default. So the most proper and simple solution would be to edit its configuration. Specifically by uncommenting the last line in this part and setting its value to `0`

```
# Set to 0 to disable, 1 to enable USB autosuspend feature.
# Default: 1

#USB_AUTOSUSPEND=1

**Reply by MartinHemma** [reply]
Thanks, I am learning new tings every day. :-)

/Martin

**Reply by Makalipay** [reply]
// Thank you for sharing a beautifully configured Debian spin. Very nice! 

* Thank you for reminding me to remove TLP as all my machines are plugged in to AC power 24/7 and additionally KDE _already_ includes the 'Powerdevil' daemon complete with a nice GUI for managing common power related things.

* In Martins case TLP is doing the exact _opposite_ of what he wants hence the need to dig around in its configuration file and _turn USB auto suspend back off again_. Per the TLP introduction "USB device plugged in
Activates USB autosuspend mode for the device (if not excluded or denylisted)."

* Gnome desktop also includes its own proprietary daemon called 'Gnome Power Manager' or something. People need to be aware that it is the _Kernel_ managing power and all these potentially conflicting software daemons can be more a hindrance than help. 

* Hence my suggestion to add a kernel parameter at boot time via grub to make sure we have power to the USB port from startup. 

Cheers

**Reply by geckolinux** [reply]
Hi @AmotanLtd , glad it was helpful.

> additionally KDE already includes the 'Powerdevil' daemon complete with a nice GUI for managing common power related things

Yep, just keep in mind that TLP works at a _much_ lower level with far more granular control over specific subsystems that consume power. Powerdevil is basically just for controlling monitor power saving and system suspend/hibernate actions. But as you mentioned, TLP might not be important or desirable for a system that is always connected to AC.
